src/padding.rs

```diff
@@ -1,12 +1,115 @@
-// Applies new padding to a BSON packet.
-// Returns the padded packet.
+use rand::Rng;
+use std::io::{Error, ErrorKind};
+
+const MIN_BYTES_PADDING: usize = 2;
+const MAX_BYTES_PADDING: usize = 32;
+
+/// Encapsulates the given BSON packet with a random number (between 2 and 32 inclusive) of random bytes before and after.
+/// # Arguments
+/// * `pkt` - A reference to a BSON packet.
+/// # Returns
+/// The encapsulated packet in Vec<u8> form.
+/// # Errors
+/// Returns an error if there is an error during the generation of random bytes or packet encapsulation
 fn encapsulate(pkt: &[u8]) -> Vec<u8> {
-    unimplemented!()
+    let mut rng = rand::thread_rng(); // use the random number generator to generate our random numbers and bytes
+    let r1 = rng.gen_range(MIN_BYTES_PADDING..=MAX_BYTES_PADDING); // length of random bytes before the BSON data
+    let r2 = rng.gen_range(MIN_BYTES_PADDING..=MAX_BYTES_PADDING); // length of random bytes after the BSON data
+
+    // vectors containing the random bytes for the front and back padding of the BSON data
+    let mut front_pad = Vec::with_capacity(r1 + 1);
+    let mut back_pad = Vec::with_capacity(r2 + 1);
+
+    while front_pad.len() < r1 {
+        let rand_byte: u8 = rng.gen();
+
+        if rand_byte != '{' as u8 {
+            front_pad.push(rand_byte);
+        }
+    }
+
+    while back_pad.len() < r2 {
+        let rand_byte: u8 = rng.gen();
+
+        if rand_byte != '}' as u8 {
+            back_pad.push(rand_byte);
+        }
+    }
+
+    front_pad.push('{' as u8);
+    back_pad.insert(0, '}' as u8);
+
+    // define a new vector to add the front pad, the packet, and finally the back pad
+    let mut encapsulated_pkt = Vec::with_capacity((pkt.len() + front_pad.len() + back_pad.len()));
+    encapsulated_pkt.extend(front_pad);
+    encapsulated_pkt.extend(pkt);
+    encapsulated_pkt.extend(back_pad);
+
+    encapsulated_pkt
 }
 
 /// Strips the padding from an unencrypted packet.
-/// Returns Ok(Vec<u8>) containing the BSON if successful.
-/// Returns Err if the packet is invalid.
+/// # Arguments
+/// * `pkt` - A reference to a BSON packet with randomized padding.
+/// # Returns
+/// The revealed BSON in Vec<u8> form if successful.
+/// # Errors
+/// Returns an error if the padding is not valid (does not have the correct braces) or
+/// if there is an error during the extraction of BSON data.
 fn reveal(pkt: &[u8]) -> Result<Vec<u8>, std::io::Error> {
-    unimplemented!()
+    // initialize bson_front_index & bson_back_index with a default of -1 to indicate they hasn't been changed
+    let mut bson_front_index: i32 = -1;
+    let mut bson_back_index = -1;
+
+    // loop through the packet and find the index of the beginning of the BSON data.
+    for (index, &element) in pkt.iter().enumerate().take(33) {
+        if element == b'{' {
+            // if the brace is not within 2-32 (inclusive) bytes, the padding is not valid
+            if index < 2 || index > 32 {
```

Comment:
minor one, but shouldnt this use `MIN_BYTES_PADDING`/`MAX_BYTES_PADDING` now that those are defined?